src/Nethermind/Nethermind.Consensus/Processing/BlockCachePreWarmer.cs

```diff
@@ -98,29 +109,36 @@ void WarmupWithdrawals(ParallelOptions parallelOptions, IReleaseSpec spec, Block
         void WarmupTransactions(ParallelOptions parallelOptions, IReleaseSpec spec, Block block, Hash256 stateRoot)
         {
             if (parallelOptions.CancellationToken.IsCancellationRequested) return;
-            Parallel.For(0, block.Transactions.Length, parallelOptions, i =>
-            {
-                // If the transaction has already been processed or being processed, exit early
-                if (block.TransactionProcessed >= i) return;
 
+            int progress = 0;
+            // We want to start at 1 which is the second transaction, giving total count one less than the length
+            Parallel.For(1, block.Transactions.Length, parallelOptions, _ =>
```

Comment:
Let's not complicate things and keep starting at 0.
Will be simpler for more changes in the future too.